contracts/modules/token/TokenController.sol

```diff
@@ -310,6 +307,62 @@ contract TokenController is ITokenController, LockHandler, MasterAwareV2 {
     }
   }
 
+  /// @notice Withdraws NXM from the Nexus platform based on specified options.
+  /// @dev Ensure the NXM is available and not locked before withdrawal. Only set flags in `WithdrawNxmOptions` for withdrawable NXM.
+  ///      Reverts if some of the NXM being withdrawn is locked or unavailable.
+  /// @param member Address of the member to withdraw NXM from.
+  /// @param stakingPoolDeposit Details for withdrawing from staking pools.
+  /// @param coverNotes Details for withdrawing V1 cover notes.
+  /// @param assessmentStake Details for withdrawing assessment stake.
+  /// @param batchSize The maximum number of iterations to avoid unbounded loops when withdrawing governance and/or assessment rewards.
+  /// @param withdrawOptions Options specifying which NXM types to withdraw, set flags to true to include specific withdrawable NXM.
+  function withdrawNXM(
```

Comment:
electing to call this `withdrawNXM` because `withdrawClaimableNXM` gives the impression that the method will only withdraw the "claimable" NXM for you.

this is not the case, the caller has the responsibility to check and pass in which NXMs are claimable